R/st_bonds.R

```diff
@@ -0,0 +1,185 @@
+#' Sensitivity analysis: `run_stress_test()` multiple times
+#'
+#' These functions performs the so called "sensitivity analysis" for the asset
+#' type in their name. They run [run_stress_test()] multiple times, iterating
+#' over multiple values of one and only one argument of [run_stress_test()].
+#'
+#' @param data Character vector of length 2. Paths to the directories
+#'   ST_INPUTS_MASTER and ST_TESTING_\{aset-type\}, e.g. ST_TESTING_BONDS,
+#'   respectively. These directories must have the files expected by the
+#'   stress-test project. The easiest way to create this object is with
+#'   [st_data_paths()].
```

Comment:
In general I believe users will appreciate not having to set env vars, but I also agree the interface for the required paths should be simple for users that are not used to R or any tech heavy work.
In that sense i like two separate input args for project_path and agnostic_path (do not use my naming here), and that the output path is simply derived from the input path (at least by default)

Reply:
For now I think it's easiest to be consistent with `run_stress_test()` and remove the `data` argument.

For later, I think CRAN forbids writing anywhere other than the session's temporary directory -- unless the user explicitely gives the path. I need to check in CRAN policy's but if that's right then we must not write data to a path hidden behind an envvar. In short, we may get away with reading data from a path in an envvar but not with writing data to it.

--

RE ease of use, I'm pretty sure I had a similar discussion for every package, and I'm sure other developers had it too. I'll find a good way to explain this and maybe write a blog post. 

If you're dealing with new users, then I think it's best to steer them in the right direction right away. Teaching them how to setup an environmental variable so they can read and write data doesn't do users a big favor. They won't be able to reuse that knowledge in the most widely used packages, e.g. the tidyverse. Soon enough R users have no choice other than learning how to read and write data from their system, and to manipulate spreadsheet-like data structure (i.e. dataframes). 

If your users don't want to do that, then they should not interact with the R package directly, but instead with a wrapper like a shiny app. The package remains complient with good design patterns, while the app gives users the interface they feel more comfirtable with.

Reply:
- remove data: alright
- CRAN forbids writing: True, so that means we  do not even have to discuss env vars further but will get rid of them very early on in CRAN ready version
- RE ease of use: I am not entirely sure I agree with the line of arguments. Users who use our code do not chose to use R and are also not intending to use if further or learn it. They only use it since we do not give them a more suitable option. So while we do not  provide them with a wrapper but force them to use the code directly I think we should put convenience for beginners very high on our priority list.